src/ol_infrastructure/components/services/vault.py

```diff
@@ -362,26 +416,34 @@ def __init__(
     ):
         super().__init__(
             "ol:services:Vault:PKI:IntermediateEnvBackendConfig",
-            backend_config.environment_name,
+            f"pki-intermediate-{backend_config.environment_name}",
             None,
             opts,
         )
 
-        resource_options = ResourceOptions(parent=self).merge(opts)  # type: ignore
+        resource_options = ResourceOptions(
+            parent=self,
```

Comment:
I'm curious what this is needed for? Not questioning the validity, just wondering how it manifests.

Reply:
I don't know about the `parent=self` thing, that was there when I got here. But the dependency on the `pki_intermediate_ca_default_issuer` is important. Pulumi wasn't able to figure out that `pki_intermediate_ca` needs to be there and complete before it can start doing _anything_ with any of the `pki_intermediate_{env}` resources.  I guess technically it could do the backends and issue the CSRs but it definitely cannot signed w/o this dependency. Seems safer to just block all `pki_intermediate_{env}` activities until `pki_intermediate_ca` is complete.

Reply:
Yeah, I remembered after posting this that the reason that's there is so that all of the resources created inside the component get properly associated with the containing component resource.